lib/http-proxy/passes/web-incoming.js

```diff
@@ -137,6 +137,12 @@ web_o = Object.keys(web_o).map(function(pass) {
     proxyReq.on('error', proxyError);
 
     function proxyError (err){
+      if (err.code === 'ECONNRESET' && proxyReq._headers && proxyReq._headers.connection === 'close') {
```

Comment:
im about to merge the case with the `econnreset` throwing errors that arent really errors. you should rebase all that and try and get the tests to pass and ill consider the use case here.